tequilapi/endpoints/service.go

```diff
@@ -189,58 +222,82 @@ func (se *ServiceEndpoint) ServiceStart(resp http.ResponseWriter, req *http.Requ
+		proposal := instance.Proposal()
+		if proposal.ProviderID == sr.ProviderID && proposal.ServiceType == sr.ServiceType {
+			return true
+		}
+	}
+	return false
+}
+
 // AddRoutesForService adds service routes to given router
-func AddRoutesForService(router *httprouter.Router, serviceManager ServiceManager) {
-	serviceEndpoint := NewServiceEndpoint(serviceManager)
+func AddRoutesForService(router *httprouter.Router, serviceManager ServiceManager, optionsParser map[string]func(json.RawMessage) (service.Options, error)) {
+	serviceEndpoint := NewServiceEndpoint(serviceManager, optionsParser)
 
 	router.GET("/services", serviceEndpoint.ServiceList)
 	router.POST("/services", serviceEndpoint.ServiceStart)
 	router.GET("/services/:id", serviceEndpoint.ServiceGet)
 	router.DELETE("/services/:id", serviceEndpoint.ServiceStop)
 }
 
-func toServiceRequest(req *http.Request) (*serviceRequest, error) {
-	var serviceRequest = serviceRequest{}
-	err := json.NewDecoder(req.Body).Decode(&serviceRequest)
-
-	return &serviceRequest, err
+func toServiceRequest(req *http.Request) (sr serviceRequest, err error) {
+	err = json.NewDecoder(req.Body).Decode(&sr)
+	return sr, err
 }
 
-func toServiceInfoResponse(state service.State) serviceInfo {
+func toServiceInfoResponse(id service.ID, instance *service.Instance) serviceInfo {
 	return serviceInfo{
-		Status: string(state),
+		ID:       string(id),
+		Status:   string(service.Running),
+		Proposal: proposalToRes(instance.Proposal()),
 	}
 }
 
-func toServiceListResponse(state service.State) serviceList {
-	return serviceList{
-		{ID: "6ba7b810-9dad-11d1-80b4-00c04fd430c8", Status: string(state)},
-		{ID: "7037d6f6-1b92-4513-8cb4-8fd1a944b996", Status: string(state)},
+func toServiceListResponse(instances map[service.ID]*service.Instance) serviceList {
+	res := make([]serviceInfo, 0)
+	for id, instance := range instances {
+		res = append(res, serviceInfo{
+			ID:       string(id),
+			Status:   string(instance.State()),
+			Proposal: proposalToRes(instance.Proposal()),
+		})
 	}
+	return res
 }
 
-func validateServiceRequest(cr *serviceRequest) *validation.FieldErrorMap {
+func validateServiceRequest(sr serviceRequest) *validation.FieldErrorMap {
 	errors := validation.NewErrorMap()
-	if len(cr.ProviderID) == 0 {
+	if len(sr.ProviderID) == 0 {
 		errors.ForField("providerId").AddError("required", "Field is required")
 	}
-	if cr.ServiceType == "" {
+	if sr.ServiceType == "" {
 		errors.ForField("serviceType").AddError("required", "Field is required")
 	}
 	return errors
 }
 
 // ServiceManager represents service manager that will be used for manipulation node services.
 type ServiceManager interface {
-	Start(providerID identity.Identity, serviceType string, options service.Options) (err error)
+	Start(providerID identity.Identity, serviceType string, options service.Options) (service.ID, error)
+	Stop(id service.ID) error
+	Service(id service.ID) *service.Instance
```

Comment:
IMHO it should be a verb. Interface provides actions.

Reply:
It can be called `GetService`, but it sounds like [it's neither idiomatic nor necessary to put Get into the getter's name](https://golang.org/doc/effective_go.html#Getters).